src/App.tsx

```diff
@@ -15,35 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ByAlphabet = 'alphabet',
+  ByLength = 'length',
+}
+
+type SortBy = {
+  sortBy: SortType | string;
+  isReverse: boolean;
+};
+
+const getPrepareGoods = (goods: string[], { sortBy, isReverse }: SortBy) => {
+  const prepareGoods: string[] = [...goods];
+
+  if (sortBy) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.ByAlphabet:
+          return good1.localeCompare(good2);
+        case SortType.ByLength:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState('');
```

Comment:
I believe you should add generic type for this state, because right now you can assign basically any string into it.